src/common/snippets/src/lowered/pass/compute_buffer_allocation_size.cpp

```diff
@@ -38,13 +42,39 @@ size_t ComputeBufferAllocationSize::get_allocation_size(const LoopManagerPtr& lo
 
     const auto& subtensor =  ov::snippets::utils::get_projected_subtensor(parent_port);
 
+    auto hard_equal = [&parent_port](const LoopPort& port) {
+        return *port.expr_port == parent_port;
+    };
+    auto soft_equal = [&](const LoopPort& loop_port) {
+        const auto& port = *loop_port.expr_port;
+        // Check semantic of LoopPort
+        if (parent_port.get_index() != port.get_index() ||
+            port.get_expr()->get_node()->get_type_info() != parent_port.get_expr()->get_node()->get_type_info())
+            return false;
+        // Check that this LoopPort is connected to the same by semantic Buffer
+        const auto consumers = port.get_connected_ports();
+        for (const auto& consumer : consumers) {
+            if (const auto buffer_consumer = ov::as_type_ptr<op::Buffer>(consumer.get_expr()->get_node())) {
+                if (buffer_consumer->get_cluster_id() == current_buffer->get_cluster_id() && consumer.get_index() == buffer_in_idx)
+                    return true;
+            }
+        }
+        return false;
+    };
+
     size_t allocation_size = 1;
     std::set<size_t> processed_dim_idxs;
     for (const auto& parent_loop : parent_loop_ids) {
         const auto loop_info = loop_manager->get_loop_info(parent_loop);
         const auto& output_ports = loop_info->get_output_ports();
-        auto it = std::find_if(output_ports.begin(), output_ports.end(), [&parent_port](const LoopPort& port) { return *port.expr_port == parent_port; });
-        OPENVINO_ASSERT(it != output_ports.end(), "compute_allocation_shape: output port of parent loop can not be found");
+        auto it = std::find_if(output_ports.begin(), output_ports.end(), hard_equal);
+        // [149219] : Try to find original loop port if this LoopInfo is cloned after InsertSpecificIterations
+        //            and ports are not mapped on the original ExpressionPorts
+        // Note: this check is needed only in Splitted Loops
+        if (it == output_ports.end()) {
```

Comment:
> If this should work only for splitted loops
>> It's not so because

What about the comment then `  // Note: this check is needed only in Splitted Loops`? Should we remove it?

Reply:
Removed it to avoid misunderstanding here